src/shell.zig

```diff
@@ -97,6 +102,18 @@ pub fn echo(shell: *Shell, comptime fmt: []const u8, fmt_args: anytype) void {
     std.debug.print(fmt_ansi, fmt_args);
 }
 
+pub fn print(shell: *Shell, comptime fmt: []const u8, fmt_args: anytype) ![]const u8 {
+    return std.fmt.allocPrint(shell.arena.allocator(), fmt, fmt_args);
+}
+
+pub fn env_get(shell: *Shell, var_name: []const u8) ![]const u8 {
+    errdefer {
+        log.err("environment variable '{s}' not defined", .{var_name});
+    }
+
+    return try std.process.getEnvVarOwned(shell.arena.allocator(), var_name);
+}
+
```

Comment:
Somewhat random utilities, I am still not entirely sure what's the desired end state for the shell is. But the guidelines are:

* convencience
* error reporting
* avoiding error-prone memory management (by allocating into shell-owned arena)